sunpy/time/time.py

```diff
@@ -212,54 +201,67 @@ def convert_time_str(time_string, **kwargs):
                 break
             if ts is None:
                 continue
-            return datetime.strptime(ts, time_format) + time_delta
+            return ap.Time.strptime(ts, time_format,
+                                    **kwargs) + time_delta
         except ValueError:
             pass
-    time_string_parse_format = kwargs.pop('_time_string_parse_format', None)
+
     if time_string_parse_format is not None:
         ts, time_delta = _regex_parse_time(time_string,
                                            time_string_parse_format)
         if ts and time_delta:
-            return datetime.strptime(ts, time_string_parse_format) + time_delta
+            return ap.Time.strptime(ts, time_string_parse_format,
+                                    **kwargs) + time_delta
         else:
-            return datetime.strptime(time_string, time_string_parse_format)
+            return ap.Time.strptime(time_string, time_string_parse_format,
+                                    **kwargs)
     # when no format matches, call default fucntion
-    convert_time.dispatch(object)(time_string, **kwargs)
+    return convert_time.dispatch(object)(time_string, **kwargs)
 
 
-def parse_time(time_string, time_format='', **kwargs):
+def parse_time(time_string, format=None, **kwargs):
     """Given a time string will parse and return a datetime object.
     Similar to the anytim function in IDL.
     utime -- Time since epoch 1 Jan 1979
+
     Parameters
     ----------
-    time_string : [ int, float, time_string, datetime ]
-        Date to parse which can be either time_string, int, datetime object.
-    time_format : [ basestring, utime, datetime ]
+    time_string : [ int, float, string, datetime, astropy.time.Time,
+                    numpy.datetime64, pandas.Timestamp ]
+        Time to parse.
+
+    format : [ 'jd', 'mjd', 'decimalyear', 'unix', 'cxcsec', 'gps',
+               'plot_date', 'datetime', 'iso', 'isot', 'yday', 'fits',
+               'byear', 'jyear', 'byear_str', 'jyear_str', 'utime']
+
```

Comment:
Should we not link to astropy docs here?

Reply:
`'utime'` is not in the docs. It is a sunpy only custom format. I can link the docs if you want.

Reply:
Can I put a link in the `format` part? Should I put it in a separate note at the end?